runtime/doc/lua.txt

```diff
@@ -3375,6 +3375,24 @@ vim.json.encode({obj}, {opts})                             *vim.json.encode()*
       • {opts}  (`table<string,any>?`) Options table with keys:
                 • escape_slash: (boolean) (default false) Escape slash
                   characters "/" in string values.
+                • pretty: (table|boolean) (default false) Enables
+                  pretty-formatting when provided or `true`. Supports the
+                  following keys:
+                  • indent: (integer|string) (default 2) Controls how nested
```

Comment:
can drop the integar variant. it's trivial to convert a number to spaces, `(' '):rep(4)`

Reply:
It's mostly about simplifying the interface and docs. When the docs get nested like this, we need to ask why.

The door is always open in the future to add more variations like this. So we don't need to have all these features right now.

This PR enables something that was previously impossible. There is no need to also add bells and whistles in this iteration.